## In our First Week, we are scrap data of Covid19 for Country-wise as well as India State-wise. In the following weeks, we are going to do EDA, get insights and deploy them in cloud

## Acquiring Data for World Cases.

### Importing necessary packages

In [1]:
import requests
from bs4 import BeautifulSoup
import numpy as np
import pandas as pd

### Getting World data 

In [2]:
url = 'https://www.worldometers.info/coronavirus/'

source = requests.get(url)
print(source)

<Response [200]>


**Response 200 denotes that the connection to the URL is successful.**

In [3]:
soup = BeautifulSoup(source.content,'html.parser')
soup.title

<title>Coronavirus Update (Live): 3,322,926 Cases and 234,408 Deaths from COVID-19 Virus Pandemic - Worldometer</title>

Verifyng the title to confirm whether we are parsing the right data

In [4]:
table = soup.find_all('table')
print(len(table))

2


In [5]:
t = table[0]

In [6]:
country = []
total_cases = []
new_cases = []
total_deaths = []
new_deaths = []
recovered = []
active_cases = []

rows = t.find_all('tr')[9:-8] #removing irrelevant columns.
#tr - parents class of the table

for row in rows:
    col = row.find_all('td')
    country.append(col[0].text.strip())
    total_cases.append(col[1].text.strip())
    new_cases.append(col[2].text.strip())
    total_deaths.append(col[3].text.strip())
    new_deaths.append(col[4].text.strip())
    recovered.append(col[5].text.strip())
    active_cases.append(col[6].text.strip())
    
print(country)
print(total_cases)
print(new_cases)
print(total_deaths)
print(new_deaths)
print(recovered)
print(active_cases)

['USA', 'Spain', 'Italy', 'UK', 'France', 'Germany', 'Turkey', 'Russia', 'Iran', 'Brazil', 'Canada', 'Belgium', 'Netherlands', 'Peru', 'India', 'Switzerland', 'Portugal', 'Ecuador', 'Saudi Arabia', 'Sweden', 'Ireland', 'Mexico', 'Singapore', 'Pakistan', 'Chile', 'Israel', 'Austria', 'Belarus', 'Japan', 'Qatar', 'UAE', 'Poland', 'Romania', 'Ukraine', 'S. Korea', 'Indonesia', 'Denmark', 'Serbia', 'Philippines', 'Bangladesh', 'Norway', 'Czechia', 'Dominican Republic', 'Australia', 'Panama', 'Colombia', 'Malaysia', 'South Africa', 'Egypt', 'Finland', 'Morocco', 'Argentina', 'Kuwait', 'Algeria', 'Moldova', 'Luxembourg', 'Kazakhstan', 'Bahrain', 'Thailand', 'Hungary', 'Greece', 'Oman', 'Afghanistan', 'Armenia', 'Iraq', 'Croatia', 'Ghana', 'Uzbekistan', 'Nigeria', 'Cameroon', 'Azerbaijan', 'Iceland', 'Bosnia and Herzegovina', 'Estonia', 'Bulgaria', 'Cuba', 'Guinea', 'New Zealand', 'North Macedonia', 'Slovenia', 'Slovakia', 'Lithuania', 'Ivory Coast', 'Bolivia', 'Djibouti', 'Hong Kong', 'Seneg

In [7]:
df = pd.DataFrame(list(zip(country,total_cases,new_cases,total_deaths,new_deaths,recovered,active_cases)),
                 columns=['Country','Total_cases','New_cases','Total_deaths','New_deaths','Recovered','Active_cases'])

df

Country Total_cases New_cases Total_deaths New_deaths  \
0                      USA   1,095,304      +281       63,871        +15   
1                    Spain     239,639                 24,543              
2                    Italy     205,463                 27,967              
3                       UK     171,253                 26,771              
4                   France     167,178                 24,376              
..                     ...         ...       ...          ...        ...   
209  Caribbean Netherlands           5                                     
210               Anguilla           3                                     
211                Comoros           1                                     
212  Saint Pierre Miquelon           1                                     
213                  China      82,874       +12        4,633              

    Recovered Active_cases  
0     155,737      875,696  
1     137,984       77,112  
2      75,945      101,551  
3         N/A      144,138  
4      49,476       93,326  
..        ...          ...  
209                      5  
210         3            0  
211                      1  
212                      1  
213    77,642          599  

[214 rows x 7 columns]

### checking the data types

In [8]:
df.dtypes

Country         object
Total_cases     object
New_cases       object
Total_deaths    object
New_deaths      object
Recovered       object
Active_cases    object
dtype: object

### We know we have na values but it's not displaying due to 'Object' datatype. NaN is considered as String

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 214 entries, 0 to 213
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Country       214 non-null    object
 1   Total_cases   214 non-null    object
 2   New_cases     214 non-null    object
 3   Total_deaths  214 non-null    object
 4   New_deaths    214 non-null    object
 5   Recovered     214 non-null    object
 6   Active_cases  214 non-null    object
dtypes: object(7)
memory usage: 11.8+ KB


In [10]:
df.columns

Index(['Country', 'Total_cases', 'New_cases', 'Total_deaths', 'New_deaths',
       'Recovered', 'Active_cases'],
      dtype='object')

## Data Cleaning

### Replacing empty/null values to NaN format

In [11]:
df.replace(r'^\s*$',np.nan,regex=True, inplace=True)
df

/home/edward/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:4172: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  method=method,


Country Total_cases New_cases Total_deaths New_deaths  \
0                      USA   1,095,304      +281       63,871        +15   
1                    Spain     239,639       NaN       24,543        NaN   
2                    Italy     205,463       NaN       27,967        NaN   
3                       UK     171,253       NaN       26,771        NaN   
4                   France     167,178       NaN       24,376        NaN   
..                     ...         ...       ...          ...        ...   
209  Caribbean Netherlands           5       NaN          NaN        NaN   
210               Anguilla           3       NaN          NaN        NaN   
211                Comoros           1       NaN          NaN        NaN   
212  Saint Pierre Miquelon           1       NaN          NaN        NaN   
213                  China      82,874       +12        4,633        NaN   

    Recovered Active_cases  
0     155,737      875,696  
1     137,984       77,112  
2      75,945      101,551  
3         N/A      144,138  
4      49,476       93,326  
..        ...          ...  
209       NaN            5  
210         3            0  
211       NaN            1  
212       NaN            1  
213    77,642          599  

[214 rows x 7 columns]

In [13]:
df.head(20)

Country Total_cases New_cases Total_deaths New_deaths Recovered  \
0            USA   1,095,304      +281       63,871        +15   155,737   
1          Spain     239,639       NaN       24,543        NaN   137,984   
2          Italy     205,463       NaN       27,967        NaN    75,945   
3             UK     171,253       NaN       26,771        NaN       N/A   
4         France     167,178       NaN       24,376        NaN    49,476   
5        Germany     163,009       NaN        6,623        NaN   126,900   
6         Turkey     120,204       NaN        3,174        NaN    48,886   
7         Russia     114,431    +7,933        1,169        +96    13,220   
8           Iran      94,640       NaN        6,028        NaN    75,103   
9         Brazil      87,187    +1,807        6,006       +105    35,935   
10        Canada      53,236       NaN        3,184        NaN    21,423   
11       Belgium      49,032      +513        7,703       +109    11,892   
12   Netherlands      39,316       NaN        4,795        NaN       N/A   
13          Peru      36,976       NaN        1,051        NaN    10,405   
14         India      35,043      +180        1,154        NaN     9,068   
15   Switzerland      29,586       NaN        1,737        NaN    23,400   
16      Portugal      25,045       NaN          989        NaN     1,519   
17       Ecuador      24,934       NaN          900        NaN     1,558   
18  Saudi Arabia      22,753       NaN          162        NaN     3,163   
19        Sweden      21,092       NaN        2,586        NaN     1,005   

   Active_cases  
0       875,696  
1        77,112  
2       101,551  
3       144,138  
4        93,326  
5        29,486  
6        68,144  
7       100,042  
8        13,509  
9        45,246  
10       28,629  
11       29,437  
12       34,271  
13       25,520  
14       24,821  
15        4,449  
16       22,537  
17       22,476  
18       19,428  
19       17,501

**We are replacing NaN values of only two columns to zero. The reason being is, we cannot fill some values based on median/mean values for these columns. It varies day to day.**

In [14]:
df.New_cases = df['New_cases'].replace({np.nan:0})
df.New_deaths = df['New_deaths'].replace({np.nan:0})

**Checking NaN values for remaining columns**

In [15]:
df.isna().sum().sort_values

<bound method Series.sort_values of Country          0
Total_cases      0
New_cases        0
Total_deaths    39
New_deaths       0
Recovered        7
Active_cases     0
dtype: int64>

**As we checked, two columns of 'Recovered' columns have N/A which is not considered as NaN. So how can we fill this data. Zero? or with some other value?. Lets do a quick math to calculate for the 3rd index row**

Total Cases: 171253
Total Deaths: 26771
Active Cases: 144138

Recovered = Total_cases - (Total_deaths + Active_cases) = 171253 - (26771 + 144138) = 344

In [17]:
df.loc[df['Recovered']=='N/A'] #344 #250

Country Total_cases New_cases Total_deaths New_deaths Recovered  \
3            UK     171,253         0       26,771          0       N/A   
12  Netherlands      39,316         0        4,795          0       N/A   

   Active_cases  
3       144,138  
12       34,271

In [21]:
df.loc[3]

Country              UK
Total_cases     171,253
New_cases             0
Total_deaths     26,771
New_deaths            0
Recovered           N/A
Active_cases    144,138
Name: 3, dtype: object

In [22]:
df.loc[3,'Recovered'] = 344
df.loc[12,'Recovered'] = 250

In [24]:
df.loc[[3,12]]

Country Total_cases New_cases Total_deaths New_deaths Recovered  \
3            UK     171,253         0       26,771          0       344   
12  Netherlands      39,316         0        4,795          0       250   

   Active_cases  
3       144,138  
12       34,271

**We have successfully replaced the values of 'N/A' in 3rd and 12th row.Lets check Whether we have to do similar imputation based on logics in some columns which have NaN**

In [25]:
df.loc[df['Total_deaths'].isna()]

Country Total_cases New_cases Total_deaths New_deaths  \
117                 Réunion         420         0          NaN          0   
126                 Vietnam         270         0          NaN          0   
128                  Rwanda         243         0          NaN          0   
131          Faeroe Islands         187         0          NaN          0   
135               Gibraltar         144         0          NaN          0   
139              Madagascar         128         0          NaN          0   
142                Cambodia         122         0          NaN          0   
151                  Uganda          83         0          NaN          0   
157              Mozambique          76         0          NaN          0   
162                     CAR          64       +14          NaN          0   
164                   Nepal          59        +2          NaN          0   
165        French Polynesia          58         0          NaN          0   
166                   Macao          45         0          NaN          0   
169                 Eritrea          39         0          NaN          0   
171                Mongolia          38         0          NaN          0   
173             South Sudan          35         0          NaN          0   
176             Timor-Leste          24         0          NaN          0   
178                 Grenada          20         0          NaN          0   
179                    Laos          19         0          NaN          0   
181                    Fiji          18         0          NaN          0   
182           New Caledonia          18         0          NaN          0   
183             Saint Lucia          17         0          NaN          0   
186                Dominica          16         0          NaN          0   
187                 Namibia          16         0          NaN          0   
188  St. Vincent Grenadines          16         0          NaN          0   
189   Saint Kitts and Nevis          15         0          NaN          0   
190              Tajikistan          15         0          NaN          0   
192        Falkland Islands          13         0          NaN          0   
197               Greenland          11         0          NaN          0   
198            Vatican City          11         0          NaN          0   
199              Seychelles          11         0          NaN          0   
203        Papua New Guinea           8         0          NaN          0   
204                  Bhutan           7         0          NaN          0   
207               St. Barth           6         0          NaN          0   
208          Western Sahara           6         0          NaN          0   
209   Caribbean Netherlands           5         0          NaN          0   
210                Anguilla           3         0          NaN          0   
211                 Comoros           1         0          NaN          0   
212   Saint Pierre Miquelon           1         0          NaN          0   

    Recovered Active_cases  
117       300          120  
126       219           51  
128       104          139  
131       184            3  
135       131           13  
139        92           36  
142       119            3  
151        52           31  
157        12           64  
162        10           54  
164        16           43  
165        50            8  
166        37            8  
169        26           13  
171        10           28  
173       NaN           35  
176        16            8  
178        13            7  
179         8           11  
181        12            6  
182        17            1  
183        15            2  
186        13            3  
187         8            8  
188         8            8  
189         6            9  
190       NaN           15  
192        13            0  
197        11            0  
198         2            9  
199         6            5  
203       N

In [26]:
df.loc[df['Recovered'].isna()]

Country Total_cases New_cases Total_deaths New_deaths  \
173            South Sudan          35         0          NaN          0   
190             Tajikistan          15         0          NaN          0   
201             MS Zaandam           9         0            2          0   
203       Papua New Guinea           8         0          NaN          0   
209  Caribbean Netherlands           5         0          NaN          0   
211                Comoros           1         0          NaN          0   
212  Saint Pierre Miquelon           1         0          NaN          0   

    Recovered Active_cases  
173       NaN           35  
190       NaN           15  
201       NaN            7  
203       NaN            8  
209       NaN            5  
211       NaN            1  
212       NaN            1

**As we checked, we haven't found any logics as like earlier. So we can replace NaN values with Zeros**

In [27]:
df.Total_deaths = df['Total_deaths'].replace({np.nan:0})
df.Recovered = df['Recovered'].replace({np.nan:0})

In [28]:
df.isna().sum().sort_values(ascending = False)

Active_cases    0
Recovered       0
New_deaths      0
Total_deaths    0
New_cases       0
Total_cases     0
Country         0
dtype: int64

**Now we have imputated Null values, its time for data cleaning '+,' symbols and convert to int datatype. Why Integer? When the data type is in object, you cannot plot different plots and get insights.**

In [29]:
df.head()

Country Total_cases New_cases Total_deaths New_deaths Recovered Active_cases
0     USA   1,095,304      +281       63,871        +15   155,737      875,696
1   Spain     239,639         0       24,543          0   137,984       77,112
2   Italy     205,463         0       27,967          0    75,945      101,551
3      UK     171,253         0       26,771          0       344      144,138
4  France     167,178         0       24,376          0    49,476       93,326

In [36]:
def plus_comma(num):
    s = str(num).replace('+','').replace(',','')
    s1 = int(s)
    return s1

In [41]:
df['New_cases'] = df['New_cases'].map(plus_comma)

In [43]:
df['New_deaths'] = df['New_deaths'].map(plus_comma)
df['Total_cases'] = df['Total_cases'].map(plus_comma)
df['Total_deaths'] = df['Total_deaths'].map(plus_comma)
df['Recovered'] = df['Recovered'].map(plus_comma)
df['Active_cases'] = df['Active_cases'].map(plus_comma)

**Converted symbols and comma using map function. Now if you check the datatype, all columns turned into integer**

In [44]:
df.head()

Country  Total_cases  New_cases  Total_deaths  New_deaths  Recovered  \
0     USA      1095304        281         63871          15     155737   
1   Spain       239639          0         24543           0     137984   
2   Italy       205463          0         27967           0      75945   
3      UK       171253          0         26771           0        344   
4  France       167178          0         24376           0      49476   

   Active_cases  
0        875696  
1         77112  
2        101551  
3        144138  
4         93326

In [45]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 214 entries, 0 to 213
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Country       214 non-null    object
 1   Total_cases   214 non-null    int64 
 2   New_cases     214 non-null    int64 
 3   Total_deaths  214 non-null    int64 
 4   New_deaths    214 non-null    int64 
 5   Recovered     214 non-null    int64 
 6   Active_cases  214 non-null    int64 
dtypes: int64(6), object(1)
memory usage: 11.8+ KB


### Feature reduction 

**New cases and New Deaths can be added with Active Cases and Total deaths respectively.**

Total Cases = Total Deaths + Recovered + Active Cases

In [46]:
df['Total_Deaths'] = df['Total_deaths'] + df['New_deaths']
df['Active_Cases'] = df['Active_cases'] + df['New_cases'] 
df.head()

Country  Total_cases  New_cases  Total_deaths  New_deaths  Recovered  \
0     USA      1095304        281         63871          15     155737   
1   Spain       239639          0         24543           0     137984   
2   Italy       205463          0         27967           0      75945   
3      UK       171253          0         26771           0        344   
4  France       167178          0         24376           0      49476   

   Active_cases  Total_Deaths  Active_Cases  
0        875696         63886        875977  
1         77112         24543         77112  
2        101551         27967        101551  
3        144138         26771        144138  
4         93326         24376         93326

In [47]:
df.drop(['New_cases','Total_deaths','New_deaths','Active_cases'],axis=1,inplace=True)
df.head()

Country  Total_cases  Recovered  Total_Deaths  Active_Cases
0     USA      1095304     155737         63886        875977
1   Spain       239639     137984         24543         77112
2   Italy       205463      75945         27967        101551
3      UK       171253        344         26771        144138
4  France       167178      49476         24376         93326

**Dropped the Old columns. Similarly we're updating the values for Total_cases**

In [48]:
df['Total_Cases'] = df['Recovered'] + df['Total_Deaths'] + df['Active_Cases']
df.drop('Total_cases',axis=1,inplace=True)
df.head()

Country  Recovered  Total_Deaths  Active_Cases  Total_Cases
0     USA     155737         63886        875977      1095600
1   Spain     137984         24543         77112       239639
2   Italy      75945         27967        101551       205463
3      UK        344         26771        144138       171253
4  France      49476         24376         93326       167178

# Acquiring Data for India Cases.

**Getting data from Official Government Website**

In [49]:
url = 'https://www.mohfw.gov.in/'

source = requests.get(url)
print(source)

<Response [200]>


**Checking whether we're getting data from the correct website**

In [50]:
soup = BeautifulSoup(source.content,'html.parser')
soup.title


<title>MoHFW | Home</title>

In [51]:
#Getting table details

table = soup.find_all('table')
print(len(table))

1


**Let's check with single content in table**

In [53]:
table[0].find('tr').text.split('\n')

['',
 'S. No.',
 'Name of State / UT',
 'Total Confirmed cases (Including 111 foreign Nationals) ',
 'Cured/Discharged/Migrated',
 'Death',
 '']

**Let's check with single content in first list in table**

In [61]:
t = table[0]
row_one = t.find_all('tr')[1:-1]
print(row_one[0].find_all('td'))
print('\n')
print(row_one[0].find_all('td')[1].text) #Getting state/UT name

[<td>1</td>, <td>Andaman and Nicobar Islands</td>, <td>33</td>, <td>16</td>, <td>0</td>]


Andaman and Nicobar Islands


**Let's do with entire table in to acquire the details**

In [62]:
state_UT = []
confirmed = []
recovered = []
death = []

rows = t.find_all('tr')[1:-1]

for row in rows[:-3]:
    col = row.find_all('td')
    state_UT.append(col[1].text.strip())
    confirmed.append(col[2].text.strip())
    recovered.append(col[3].text.strip())
    death.append(col[4].text.strip())
    
print(state_UT)
print(confirmed)
print(recovered)
print(death)

['Andaman and Nicobar Islands', 'Andhra Pradesh', 'Arunachal Pradesh', 'Assam', 'Bihar', 'Chandigarh', 'Chhattisgarh', 'Delhi', 'Goa', 'Gujarat', 'Haryana', 'Himachal Pradesh', 'Jammu and Kashmir', 'Jharkhand', 'Karnataka', 'Kerala', 'Ladakh', 'Madhya Pradesh', 'Maharashtra', 'Manipur', 'Meghalaya', 'Mizoram', 'Odisha', 'Puducherry', 'Punjab', 'Rajasthan', 'Tamil Nadu', 'Telengana', 'Tripura', 'Uttarakhand', 'Uttar Pradesh', 'West Bengal']
['33', '1463', '1', '42', '426', '56', '40', '3515', '7', '4395', '313', '40', '614', '111', '576', '497', '22', '2719', '10498', '2', '12', '1', '143', '8', '357', '2584', '2323', '1039', '2', '57', '2281', '795']
['16', '403', '1', '29', '82', '17', '36', '1094', '7', '613', '209', '28', '216', '20', '235', '383', '16', '482', '1773', '2', '0', '0', '41', '5', '90', '836', '1258', '441', '2', '36', '555', '139']
['0', '33', '0', '1', '2', '0', '0', '59', '0', '214', '3', '1', '8', '3', '22', '4', '0', '137', '459', '0', '1', '0', '1', '0', '19', '5

**Construct a dataframe with the scrapped details**

In [63]:
df_india = pd.DataFrame(list(zip(state_UT,recovered,death,confirmed)),columns=['State','Recovered','Total_Deaths',
                                                                              'Total_Cases'])
df_india.head()

State Recovered Total_Deaths Total_Cases
0  Andaman and Nicobar Islands        16            0          33
1               Andhra Pradesh       403           33        1463
2            Arunachal Pradesh         1            0           1
3                        Assam        29            1          42
4                        Bihar        82            2         426

**Let's verify the datatype before proceeding for data cleaning**

In [64]:
df_india.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   State         32 non-null     object
 1   Recovered     32 non-null     object
 2   Total_Deaths  32 non-null     object
 3   Total_Cases   32 non-null     object
dtypes: object(4)
memory usage: 1.1+ KB


**Convert the Object datatypes to Integer**

In [65]:
cols = ['Total_Cases','Recovered','Total_Deaths']

for col in cols:
    df_india[col] = df_india[col].astype(int)

In [66]:
df_india.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   State         32 non-null     object
 1   Recovered     32 non-null     int64 
 2   Total_Deaths  32 non-null     int64 
 3   Total_Cases   32 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 1.1+ KB


**Generate Active Cases from the existing features**

In [67]:
df_india['Active_Cases'] = df_india['Total_Cases'] - (df_india['Recovered'] + df_india['Total_Deaths'])
df_india.head()

State  Recovered  Total_Deaths  Total_Cases  \
0  Andaman and Nicobar Islands         16             0           33   
1               Andhra Pradesh        403            33         1463   
2            Arunachal Pradesh          1             0            1   
3                        Assam         29             1           42   
4                        Bihar         82             2          426   

   Active_Cases  
0            17  
1          1027  
2             0  
3            12  
4           342

In [69]:
df.head() #From Corona - World Dataset

Country  Recovered  Total_Deaths  Active_Cases  Total_Cases
0     USA     155737         63886        875977      1095600
1   Spain     137984         24543         77112       239639
2   Italy      75945         27967        101551       205463
3      UK        344         26771        144138       171253
4  France      49476         24376         93326       167178

### As we check our dataframes df and df_india columns are in different position. Lets reorder so that we'd merge in the upcoming week and do data analysis

In [70]:
df_india = df_india[['State','Recovered','Total_Deaths','Active_Cases','Total_Cases']]
df_india.head()

State  Recovered  Total_Deaths  Active_Cases  \
0  Andaman and Nicobar Islands         16             0            17   
1               Andhra Pradesh        403            33          1027   
2            Arunachal Pradesh          1             0             0   
3                        Assam         29             1            12   
4                        Bihar         82             2           342   

   Total_Cases  
0           33  
1         1463  
2            1  
3           42  
4          426

**Lets do a sample check in India Coronavirus data. Top 5 Total_deaths state**

In [71]:
df_india.sort_values('Total_Deaths',ascending=False).head()

State  Recovered  Total_Deaths  Active_Cases  Total_Cases
18     Maharashtra       1773           459          8266        10498
9          Gujarat        613           214          3568         4395
17  Madhya Pradesh        482           137          2100         2719
7            Delhi       1094            59          2362         3515
25       Rajasthan        836            58          1690         2584

In [72]:
df.sort_values('Total_Deaths',ascending=False).head()

Country  Recovered  Total_Deaths  Active_Cases  Total_Cases
0     USA     155737         63886        875977      1095600
2   Italy      75945         27967        101551       205463
3      UK        344         26771        144138       171253
1   Spain     137984         24543         77112       239639
4  France      49476         24376         93326       167178